src/model/dfcxx/lib/dfcxx/IRbuilders/builder.cpp

```diff
@@ -7,60 +7,280 @@
+      newOp = builder.create<mlir::dfcir::NegOp>(loc, conv[node.var],
+                                                 map[first]);
+      break;
+    }
+    case LESS: {
+      Node first = ins[0].source;
+      Node second = ins[1].source;
+      newOp = builder.create<mlir::dfcir::LessOp>(loc, conv[node.var],
+                                                  map[first], map[second]);
+      break;
+    }
+    case LESSEQ: {
+      Node first = ins[0].source;
+      Node second = ins[1].source;
+      newOp = builder.create<mlir::dfcir::LessEqOp>(loc, conv[node.var],
+                                                    map[first],
+                                                    map[second]);
+      break;
+    }
+    case GREATER: {
+      Node first = ins[0].source;
+      Node second = ins[1].source;
+      newOp = builder.create<mlir::dfcir::GreaterOp>(loc, conv[node.var],
+                                                     map[first],
+                                                     map[second]);
+      break;
+    }
+    case GREATEREQ: {
+      Node first = ins[0].source;
+      Node second = ins[1].source;
+      newOp = builder.create<mlir::dfcir::GreaterEqOp>(loc, conv[node.var],
+                                                       map[first],
+                                                       map[second]);
+      break;
+    }
+    case EQ: {
+      Node first = ins[0].source;
+      Node second = ins[1].source;
+      newOp = builder.create<mlir::dfcir::EqOp>(loc, conv[node.var],
+                                                map[first],
+                                                map[second]);
+      break;
+    }
+    case NEQ: {
+      Node first = ins[0].source;
+      Node second = ins[1].source;
+      newOp = builder.create<mlir::dfcir::NotEqOp>(loc, conv[node.var],
+                                                   map[first],
+                                                   map[second]);
+      break;
+    }
+    case SHL: {
+      Node first = ins[0].source;
+      auto attrType = mlir::IntegerType::get(builder.getContext(), 32,
+                                             mlir::IntegerType::Signless);
+      auto attr = mlir::IntegerAttr::get(attrType, node.data.bitShift);
+                                  
+      newOp = builder.create<mlir::dfcir::ShiftLeftOp>(loc, conv[node.var],
+                                                       map[first], attr);
+      break;
+    }
+    case SHR: {
+      Node first = ins[0].source;
+      auto attrType = mlir::IntegerType::get(builder.getContext(), 32,
+                                             mlir::IntegerType::Signless);
+      auto attr = mlir::IntegerAttr::get(attrType, node.data.bitShift);
+                                  
+      newOp = builder.create<mlir::dfcir::ShiftRightOp>(loc, conv[node.var],
+                                                        map[first], attr);
+      break;
+    }
+    default: assert(false);

```

Comment:
It will be helpful to print an informative message that shows unexpected 'node.type' value